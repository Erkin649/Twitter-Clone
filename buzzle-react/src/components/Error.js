import React from 'react';

const Error = ({ error, clearError }) =>
    error ?
    <div className="row justify-content-center mt-3">
        <div className="col-md-6">
            <div className="alert alert-danger">
                {error}
                <button type="button" className="close" onClick={clearError}>
                    <span>&times;</span>
                </button>
            </div>
        </div>
    </div>
    :
    <></>

export default Error;
