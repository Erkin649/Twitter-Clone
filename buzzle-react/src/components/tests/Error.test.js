import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Error from '../Error.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Error />, div);
});

it('renders without crashing with error', () => {
    const wrapper = shallow(<Error error="Something went wrong..." />);
    expect(wrapper.contains("Something went wrong...")).toBe(true);
});

it('renders correctly and equal to snapshot', () => {
    const tree = renderer.create(<Error error="Something went wrong..." />)
                         .toJSON();

    expect(tree).toMatchSnapshot();
});
