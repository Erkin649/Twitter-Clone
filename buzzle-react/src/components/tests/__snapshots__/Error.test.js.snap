// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and equal to snapshot 1`] = `
<div
  className="row justify-content-center mt-3"
>
  <div
    className="col-md-6"
  >
    <div
      className="alert alert-danger"
    >
      Something went wrong...
      <button
        className="close"
        type="button"
      >
        <span>
          Ã—
        </span>
      </button>
    </div>
  </div>
</div>
`;
